newlib/libc/sys/hipperos/include/sys/_pthreadtypes.h

```diff
@@ -115,9 +124,35 @@ typedef struct {
 #endif
     int recursive;
 } pthread_mutexattr_t;
-#endif /* !defined(__XMK__) */
 
-#define _PTHREAD_MUTEX_INITIALIZER ((pthread_mutex_t) 0xFFFFFFFF)
+#if defined(_UNIX98_THREAD_MUTEX_ATTRIBUTES)
+#define _PTHREAD_MUTEX_INITIALIZER     \
+    ((pthread_mutex_t){                \
+        .type = PTHREAD_MUTEX_DEFAULT, \
+        .is_initialized = 1,           \
+        .owner = 0xFFFFFFFFu,          \
+        .lock_count = 0u,              \
+    })
+#else
+#define _PTHREAD_MUTEX_INITIALIZER     \
```

Comment:
The `else  is useless, right?

Reply:
For the current implementation, yes. But as we have ifdefs for the mutex types above, this makes sense.